internal/epoll/poller.go

```diff
@@ -118,8 +119,8 @@ func (p *Poller) Add(fd int, id int) error {
 // Wait for events.
 //
 // Returns the number of pending events or an error wrapping os.ErrClosed if
-// Close is called.
-func (p *Poller) Wait(events []unix.EpollEvent) (int, error) {
+// Close is called, or os.ErrDeadlineExceeded if EpollWait timeout.
+func (p *Poller) Wait(events []unix.EpollEvent, deadline time.Time) (int, error) {
```

Comment:
From a personal point of view, I would prefer `time.Duration` over `time.Time` as argument type of `deadline`.
Is there a use case to specify a date like `2022-08-01 00:00:00 +0000 UTC` as deadline?

Reply:
That would be SetTimeout. I mentioned this in the other PR: we can add SetTimeout as a convenience wrapper around SetDeadline, but the underlying implementation must use time.Time.

Reply:
Or do you mean Duration here and keep time.Time for the public API?

Reply:
Setting a fixed date (month/day/year ...) in the future (or past) as a deadline to read from a socket feels strange to me. Usually something like `s`, `ms` or `ns` is used. But maybe that is just my usage of sockets.

even for the public API I would not expect so see a `time.Time` as input for such a functionality.

Reply:
yeah make sense. `SetDeadline(time.Time)` and `SetTimeout(Duration)`